hera_cal/io.py

```diff
@@ -658,25 +658,42 @@ def load_flags(flagfile, filetype='h5', return_meta=False):
         from pyuvdata import UVFlag
         uvf = UVFlag(flagfile)
         assert uvf.mode == 'flag', 'The input h5-based UVFlag object must be in flag mode.'
-        assert np.issubdtype(uvf.polarization_array.dtype, np.signedinteger), \
-            "The input h5-based UVFlag object's polarization_array must be integers."
+        assert (np.issubsctype(uvf.polarization_array.dtype, np.signedinteger)
+                or np.issubsctype(uvf.polarization_array.dtype, np.bites_)), \
+            "The input h5-based UVFlag object's polarization_array must be integers or byte strings."
         freqs = np.unique(uvf.freq_array)
         times = np.unique(uvf.time_array)
         history = uvf.history
-        pol_indices = {polnum2str(polnum): i for i, polnum in enumerate(uvf.polarization_array)}
+
         if uvf.type == 'baseline':  # one time x freq waterfall per baseline
             blt_slices = get_blt_slices(uvf)
-            for ip, polnum in enumerate(uvf.polarization_array):
+            for ip, pol in enumerate(uvf.polarization_array):
+                if np.issubdtype(uvf.polarization_array.dtype, np.signedinteger):
+                    pol = polnum2str(pol)  # convert to string if possible
+                else:
+                    pol = ','.join([polnum2str(int(p)) for p in pol.split(b',')])
```

Comment:
What do you suggest? Should I assert that it has a comma if it it's a string?

Reply:
I was thinking having a try and except. It doesn't appear this transformation is required for downstream (it's just a key in a dictionary right?), so maybe if `polnum2str` fails, it resorts to just keeping `pol` as-is? Or would that be too much of a hassle downstream?

Reply:
I kinda want to keep it as an error so that if we ever change the standard we'll be forced to reevaluate what we're doing here. There's no downstream issues, since this code isn't really used in the pipeline.